--- a/dlls/msvcrt/string.c
+++ b/dlls/msvcrt/string.c
@@ -3182,7 +3182,7 @@
 }


-#if defined(__i386__) || defined(__x86_64__)
+#if defined(__i386__) || (defined(__x86_64__) && !defined(__arm64ec__))

 #ifdef __i386__
 #define DEST_REG "%edi"
@@ -3295,7 +3295,7 @@
         if (n <= 64) return dst;

         n = (n - a) & ~0x1f;
-#if defined(__i386__) || defined(__x86_64__)
+#if defined(__i386__) || (defined(__x86_64__) && !defined(__arm64ec__))
         if (n >= 2048 && erms_supported)
         {
             erms_memset_aligned_32(d + a, v, n);
@@ -3312,7 +3312,7 @@
         }
 #endif
 #endif
-#ifndef __x86_64__
+#if !defined(__x86_64__) || defined(__arm64ec__)
         memset_aligned_32(d + a, v, n);
         return dst;
 #endif
